google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -3547,4 +3547,43 @@ HmacKeyMetadata updateHmacKeyState(
    * @throws StorageException upon failure
    */
   ServiceAccount getServiceAccount(String projectId);
+
+  /**
+   * Creates a notification subscription for a given bucket.
+   *
+   * @param bucket name of the bucket
+   * @param notification a {@code Notification} object
+   * @return the notification that was created
+   * @throws StorageException upon failure
+   */
+  Notification createNotification(String bucket, Notification notification);
```

Comment:
I can't understand the logic here: what is the difference between the input notification and the returned one.
Intuitively I think that this method is about to subscribe. If so it should look like:

`Notification subscribe(String bucket, SubscriptionRequest request)` or may be:
`Notification subscribe(String bucket, String topic)`

Reply:
@dmitry-fa most of the implementation pull out from this artifact https://github.com/Googleapis/java-notification.
It was implemented in a separate artifact and customers would benefit from having it in the same artifact.

Reply:
The major problem here not in the method name, but in the description. I can't understand how I can use this new API.

New introduced entity `Notification` is described as:

```
/**
 * Google Storage Notification metadata;
 *
 * @see <a href="https://cloud.google.com/storage/docs/concepts-techniques#concepts">Concepts and
 *     Terminology</a>
 */
```
The link points to a page which doesn't mention 'Notification', so this link is not very helpful.

`createNotification` name sounds confusing to me. Probably it will sound confusing to others.  A few more words describing how to use this method will be useful. An example of use would be awesome.

Reply:
I think the right name for the method is `addNotification()`
Actually this method adds a new notification to the list of bucket's notifications.
I guess the difference between input and output notifications - the output has the server-generated id set.

Reply:
we could give that name as well but according to api doc description `createNotification `looks more informative here.

Reply:
createNotification is the internal name for an RPC call. Methods in the Storage interface could be named independently.

Reply:
done